windev.ps1

```diff
@@ -1,55 +1,181 @@
     }
 
-    $script = Invoke-RestMethod $url
-    # Elevate Shell if necessary
-    if (!([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
-        Write-Output "Winutil needs to be run as Administrator. Attempting to relaunch."
+    return $WinUtilDownloadURL
+}
+
+# Get the URL used to download the latest release of WinUtil from the releases page.
+$WinUtilReleaseURL = Get-WinUtilReleaseURL
 
-        $powershellcmd = if (Get-Command pwsh -ErrorAction SilentlyContinue) { "pwsh" } else { "powershell" }
-        $processCmd = if (Get-Command wt.exe -ErrorAction SilentlyContinue) { "wt.exe" } else { $powershellcmd }
+# Create the file path that the latest WinUtil release will be downloaded to on the local disk.
+$WinUtilScriptPath = Join-Path "$env:TEMP" "winutil.ps1"
 
-        Start-Process $processCmd -ArgumentList "$powershellcmd -ExecutionPolicy Bypass -NoProfile -Command $(Invoke-Expression $script)" -Verb RunAs
+# Function to download the latest release of WinUtil from the releases page to the local disk.
+function Get-LatestWinUtil {
+    if (!(Test-Path $WinUtilScriptPath)) {
+        Invoke-RestMethod $WinUtilReleaseURL -OutFile $WinUtilScriptPath
     }
-    else{
-        Invoke-Expression $script
+}
+
+# Function to check for any version changes to WinUtil, re-downloading the script if it has been upgraded/downgraded.
+function Get-WinUtilUpdates {
+    # Make a web request to the latest WinUtil release URL and store the script's raw code for processing.
+    $RawWinUtilScript = (Invoke-WebRequest $WinUtilReleaseURL -UseBasicParsing).RawContent
+
+    # Create the regular expression pattern used to extract the script's version number from the script's raw content.
+    $VersionExtractionRegEx = "(\bVersion\s*:\s)([\d.]+)"
+
+    # Match the entire 'Version:' header and extract the script's version number directly using RegEx capture groups.
+    $RemoteWinUtilVersion = (([regex]($VersionExtractionRegEx)).Match($RawWinUtilScript).Groups[2].Value)
+    $LocalWinUtilVersion = (([regex]($VersionExtractionRegEx)).Match((Get-Content $WinUtilScriptPath)).Groups[2].Value)
+
+    # If WinUtil has been modified since the last time it was launched, download a fresh copy of the script.
+    if ([version]$RemoteWinUtilVersion -ne [version]$LocalWinUtilVersion) {
+        Write-Host "WinUtil is out-of-date. Downloading WinUtil '$($RemoteWinUtilVersion)'..." -ForegroundColor DarkYellow
+        Invoke-RestMethod $WinUtilReleaseURL -OutFile $WinUtilScriptPath
+    }
+
+    # If WinUtil has not been modified, skip updating the script and let the user know it is already up-to-date. 
+    if ([version]$RemoteWinUtilVersion -eq [version]$LocalWinUtilVersion) {
+        Write-Host "WinUtil is already up-to-date. Skipped update checking." -ForegroundColor Yellow
+    }
```

Comment:
Since these are mutually exclusive conditions, this can be simplified with `else` like so:
```suggestion
# Check if WinUtil needs an update and either download a fresh copy or notify the user.
if ([version]$RemoteWinUtilVersion -ne [version]$LocalWinUtilVersion) {
    Write-Host "WinUtil is out-of-date. Downloading WinUtil '$RemoteWinUtilVersion'..." -ForegroundColor DarkYellow
    Invoke-RestMethod $WinUtilReleaseURL -OutFile $WinUtilScriptPath
} else {
    Write-Host "WinUtil is already up-to-date. Skipped update checking." -ForegroundColor Yellow
}
```